contracts/ArrowVestingFactory.sol

```diff
@@ -10,9 +10,16 @@ import "./ArrowVestingBase.sol";
  */
 contract ArrowVestingFactory {
 
-    address immutable vestingImplementation;
+    address public vestingImplementation;
```

Comment:
What does changing this to `public` achieve? I believe that reading `immutable` variables are cheaper from a gas POV than `public`.

Reply:
immutable means we cannot change it, thus making the implementation un-upgradeable. Must be public in my opinion for the market to see it and for us to upgrade